src/main/java/it/gov/pagopa/pu/pagopapayments/service/printpaymentnotice/GenerateNoticeService.java

```diff
@@ -63,6 +63,23 @@ public GeneratedNoticeMassiveFolderDTO generateNoticeMassive(NoticeRequestMassiv
     return GeneratedNoticeMassiveFolderMapper.toGeneratedNoticeMassiveFolderDTO(response);
   }
 
+  public SignedUrlResultDTO getNoticeMassiveZip(Long organizationId, String folderId, String accessToken) {
+    Organization org = organizationService.getOrganizationById(organizationId, accessToken);
+    GetGenerationRequestStatusResourceDTO folderStatus = printPaymentNoticeService.getFolderStatus(org.getBrokerId(), folderId, accessToken);
+    SignedUrlResultDTO result = new SignedUrlResultDTO();
+
+    GetGenerationRequestStatusResourceDTO.StatusEnum status = folderStatus.getStatus();
+    if (status == PROCESSED || status == PROCESSED_WITH_FAILURES || status == FAILED) {
```

Comment:
let's always use equals method.
here you could also use a constant Set instead of writing all the OR equals conditions
```suggestion
    if (PROCESSED.equals(status) || PROCESSED_WITH_FAILURES.equals(status) || FAILED.equals(status)) {
```